crates/ty_python_semantic/resources/mdtest/narrow/conditionals/eq.md

```diff
@@ -35,11 +41,31 @@ class Answer(Enum):
 
 def _(answer: Answer):
     if answer != Answer.NO:
-        # TODO: This should be simplified to `Literal[Answer.YES]`
-        reveal_type(answer)  # revealed: Answer & ~Literal[Answer.NO]
+        reveal_type(answer)  # revealed: Literal[Answer.YES]
+    else:
+        reveal_type(answer)  # revealed: Literal[Answer.NO]
+
+def _(answer: Answer):
+    if answer == Answer.NO:
```

Comment:
I think we need to forbid custom `__eq__` on enum types in order to make this sound. Or maybe mark enum types with custom `__eq__` and refuse to do equality narrowing on them? (I'm fine with this just being an issue for later follow-up, if anyone ever cares enough.)

```
>>> class Color(Enum):
...     RED = 1
...     BLUE = 2
...     def __eq__(self, other):
...         return True
...
>>> Color.RED == Color.BLUE
True
>>> Color.RED == 2
True
>>> Color.RED == 7
True
```